sdk/include/psdk/appmodel.h

```diff
@@ -0,0 +1,72 @@
+
+#pragma once
+
+typedef enum AppPolicyProcessTerminationMethod
+{
+    AppPolicyProcessTerminationMethod_ExitProcess = 0,
+    AppPolicyProcessTerminationMethod_TerminateProcess = 1,
+} AppPolicyProcessTerminationMethod;
+
+typedef enum AppPolicyThreadInitializationType
+{
+    AppPolicyThreadInitializationType_None = 0,
+    AppPolicyThreadInitializationType_InitializeWinRT = 1,
+} AppPolicyThreadInitializationType;
+
+typedef enum AppPolicyShowDeveloperDiagnostic
+{
+    AppPolicyShowDeveloperDiagnostic_None = 0,
+    AppPolicyShowDeveloperDiagnostic_ShowUI = 1,
+} AppPolicyShowDeveloperDiagnostic;
+
+typedef enum AppPolicyWindowingModel
+{
+    AppPolicyWindowingModel_None = 0,
+    AppPolicyWindowingModel_Universal = 1,
+    AppPolicyWindowingModel_ClassicDesktop = 2,
+    AppPolicyWindowingModel_ClassicPhone = 3
+} AppPolicyWindowingModel;
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
Shouldn't the typedef'ed enums be inside the extern C as well?

Reply:
POD types do not have a "linkage", so it makes no difference. It is different for classes with member functions and template types, which need to be `extern "C++"`.